stmtools/_io.py

```diff
@@ -0,0 +1,137 @@
+      three variables: `space_pattern`, `spacetime_pattern` and `coords_cols`.
+
+    The following assumptions are made to the column names of the csv file:
+        1. All columns with space-only attributes share the same RE pattern in the column names.
+          E.g. Lattitude, Longitude and height columns are named as "pnt_lat", "pnt_lon" and 
+          "pnt_height", sharing the same RE pattern "^pnt_";
+        2. Per space-time attribute, a common RE pattern is shared by all columns. E.g. for the 
+          time-sries of amplitude data, the column names are "amp_20100101", "amp_20100110", 
+          "amp_20100119" ..., where "^amp_" is the common RE pattern;
+        3. There is no temporal-only (i.e. 1-row attribute) attribute presented the csv file.
+
+    `from_csv` does not retrieve time stamps based on column names. The `time` coordinate of 
+      the output STM will be a monotonic integer series starting from 0.
+
+    Args:
+        file (str | Path): Path to the csv file.
+        space_pattern (str, optional): RE pattern to match space attribute columns. 
+          Defaults to "^pnt_".
+        spacetime_pattern (dict | None, optional): A dictionay mapping RE patterns of each 
+          space-time attribute to corresponding variable names. Defaults to None, which means
+          the following map will be applied: 
+          {"^d_": "deformation", "^a_": "amplitude", "^h2ph_": "h2ph"}.
+        coords_cols (list | dict, optional): List of columns to be used as space coordinates. 
+          When `coords_cols` is a dictionary, a reaming will be performed per coordinates. 
+          Defaults to None, then the following renaming will be performed: 
+          "{"pnt_lat": "lat", "pnt_lon": "lon"}"
+        output_chunksize (dict | None, optional): Chunksize of the output. Defaults to None, 
+          then the size of the first chunk in the DaskDataFrame will be used, up-rounding to 
+          the next 5000.
+        blocksize (int | str | None, optional): Blocksize to load the csv. 
+          Defaults to 200e6 (in bytes). See the documentation of 
+          [dask.dataframe.read_csv](https://docs.dask.org/en/stable/generated/dask.dataframe.read_csv.html)
+
+    Returns:
+        xr.Dataset: Output STM instance
+    """
+
+    # Load csv as Dask DataFrame
+    ddf = dd.read_csv(file, blocksize=blocksize)
+
+    # Take the first column and compute the chunk sizes
+    da_col0 = ddf[ddf.columns[0]].to_dask_array()
+    da_col0.compute_chunk_sizes()
+    chunks = da_col0.chunks
+
+    # Count time dimension by one spacetime column
+    if spacetime_pattern is None:
+        spacetime_pattern = {"^d_": "deformation", "^a_": "amplitude", "^h2ph_": "h2ph"}
+    time_shape = 0
+    if spacetime_pattern is not None:
+        key = list(spacetime_pattern.keys())[0]
+        for column in ddf.columns:
+            if re.match(re.compile(key), column):
```

Comment:
I think it would be useful for users to know if the pattern in csv files is valid. So perhaps adding `else` statement with a clear error message will help.

Reply:
Thanks for the suggestion! I added on check at the beginning of the function, to check if all specified space/space-time patterns have at least one match. Otherwise an ValueError is raised